### PR TITLE
Adding PSInputNumber component

### PR DESCRIPTION
**Pull request by NeOMakinG**
Related to https://github.com/PrestaShop/PrestaShop/issues/16599

https://github.com/PrestaShop/prestashop-ui-kit/pull/70/ needs to be merged before, and this PR needs to be updated afterwhile

How to test : Build assets (npm run build or npm run watch) and then compare with issue's specs on the index.html of the UI Kit, if there are var missing, it's because past PR's needs to be merged before..

### CODE REVIEW

File: js/components/ps-number-input.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    this.labelMin = element.dataset.labelMin;
+    this.labelMax = element.dataset.labelMax;
+    this.labelNan = element.dataset.labelNan;
+    this.incrementButton = element.querySelector('.ps-number-increment');
+    this.decrementButton = element.querySelector('.ps-number-decrement');
+    this.input = element.querySelector('input');
+    this.invalidElement = element.querySelector('.invalid-feedback');
+
+    this.initValue = () => {
+        if(this.input.value.replace('.', '') !== this.input.value || this.input.value.replace(',', '') !== this.input.value) {
```
Comment (PierreRambaud) [inline js/components/ps-number-input.js]: You should add eslint :thinking:

File: js/components/ps-numbers.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Permit to init every PSInputNumber object on the page
+ *
+ * @param {string} selector
+ * @method init
+ * @returns {Object}
+ */
+const PSNumbers = function PSNumbers(selector) {
+    this.selector = selector;
```
Comment (PierreRambaud) [inline js/components/ps-numbers.js]: Since it's a function and not a class, you shouldn't use  `this`

File: js/components/ps-number-input.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Permit to create an input number element as PrestaShop wants it
+ *
+ * @param {DOM Element Object} element
+ * @method validate
+ * @method updateValue
+ * @returns {Object}
+ */
+const PSNumberInput = function PSNumberInput(element) {
+    this.min = Number(element.dataset.min);
```
Comment (PierreRambaud) [inline js/components/ps-number-input.js]: If you want to use `this` use a class, not a function :thinking:

File: js/prestashop-ui-kit.js [hunk 4]
```diff
@@ -167,6 +168,7 @@ $.fn.pstooltip = $.fn.tooltip;
         initToolTips();
         initSearchBar();
         initInputFile();
+        new PSNumbers('.ps-number');
```
Comment (PierreRambaud) [inline js/prestashop-ui-kit.js]: It's not an object :thinking:

File: js/components/ps-number-input.js (hunk 1 above)
Comment (NeOMakinG) [reply js/components/ps-number-input.js]: It will be a part of another PR, not here

File: js/components/ps-number-input.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  const labelNan = element.dataset.labelNan;
+  let incrementButton = element.querySelector(".ps-number-increment");
+  let decrementButton = element.querySelector(".ps-number-decrement");
+  let input = element.querySelector("input");
+  let invalidElement = element.querySelector(".invalid-feedback");
+  let value = input.value;
+
+  let initValue = () => {
+    if (
+      input.value.replace(".", "") !== input.value ||
```
Comment (PierreRambaud) [inline js/components/ps-number-input.js]:
```suggestion
      input.value.replace('.', '') !== input.value ||
```
single quote :)

**Comment by PierreRambaud** [comment]
Thanks @NeOMakinG

**Pull request merged**